src/iproject.ts

```diff
@@ -17,22 +17,82 @@ import { IBMiObject } from "@halcyontech/vscode-ibmi-types";
 import { ProjectManager } from "./projectManager";
 import { ValidatorResult } from "jsonschema";
 
-const DEFAULT_CURLIB = '&CURLIB';
-const DEFAULT_OBJLIB = '&OBJLIB';
+/**
+ * Represents the default variable for a project's current library.
+ */
+export const DEFAULT_CURLIB = '&CURLIB';
+
+/**
+ * Represents the default variable for a project's object library.
+ */
+export const DEFAULT_OBJLIB = '&OBJLIB';
 
+/**
+ * Represents a file that stores project information.
+ */
 export type ProjectFileType = 'iproj.json' | '.ibmi.json' | 'joblog.json' | 'output.log' | '.env';
+
+/**
+ * Represents a project's library list.
+ */
 export type LibraryList = { libraryInfo: IBMiObject; libraryType: string; }[];
+
+/**
+ * Represents the environment variables in a `.env` file.
+ */
 export type EnvironmentVariables = { [name: string]: string };
+
+/**
+ * Represents the direction an entry should be moved.
+ */
 export type Direction = 'up' | 'down';
+
+/**
+ * Represents the position of an entry.
+ */
 export type Position = 'first' | 'last' | 'middle';
 
+/**
+ * Represents an IBM i Project.
+ */
 export class IProject {
+  /**
+   * The project's name.
+   */
   private name: string;
+
+  /**
+   * An object that represents the contents of the project's `iproj.json` with
+   * variables resolved based on the project's `.env` file.
+   */
   private state: IProjectT | undefined;
+
+  /**
+   * A map of directory locations and the content of each `.ibmi.json` file in
+   * a project. *Note* that for the root `.ibmi.json` file, the contents of the
+   * `iproj.json` file will take precedence.
```

Comment:
Let us know if the Type has variables resolved or not

Reply:
The variables are not resolved in the buildMap. I have updated the TS doc